tasks/datasources.yml

```diff
@@ -0,0 +1,60 @@
+---
+
+# Get existing datasources
+- name: datasources > Retrieve datasources
+  uri:
+    url:              "{{ manala_grafana_api_url }}/api/datasources"
+    force_basic_auth: true
+    user:             "{{ manala_grafana_api_user }}"
+    password:         "{{ manala_grafana_api_password }}"
+    return_content:   true
+  register: _manala_grafana_datasources_output
+
+# Update existing datasources
+- name: datasources > Update datasources
+  register: _manala_grafana_datasources_update_output
+  failed_when: "'Datasource updated' not in _manala_grafana_datasources_update_output|to_json"
+  uri:
+    url:                  "{{ manala_grafana_api_url }}/api/datasources/{{ item[0].id }}"
+    method:               PUT
+    HEADER_Content-Type:  "application/json"
+    body:                 "{{ item[1]|to_json }}"
+    force_basic_auth:     true
+    user:                 "{{ manala_grafana_api_user }}"
+    password:             "{{ manala_grafana_api_password }}"
+    status_code:          200
+  with_nested:
+    - "{{ _manala_grafana_datasources_output.json }}"
```

Comment:
```diff
-    - "{{ _manala_grafana_datasources_output.json }}"
+    - "{{ _manala_grafana_datasources_output.json|default([]) }}"
```

Will avoid deprecated warning :
> TASK [manala.grafana : datasources > Update datasources] ***********************
[DEPRECATION WARNING]: Skipping task due to undefined Error, in the future this will be a fatal error.: 'dict object' has no attribute 'json'.
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.